src/app/(main)/blog/articles/api-authentication-techniques.md

```diff
@@ -0,0 +1,132 @@
+
+
+### What is API Authentication
+
+Simply put, API authentication refers to the act of restricting access to API resources to only trusted entities. To determine a trusted entity, an API server would expect an entity to meet certain criteria, such as possessing a valid credential. Authentication here is closely related to authorization, but their roles differ. API authorization would refer to a means of specifying what an entity or user can or cannot do with an API even after they have gained access to it.
+
+
+### The least common Auth Technique: Basic Authentication
+
+HTTP Basic Authentication is the least secure among all the mechanisms for API authentication, hence it is the least adopted. This form of authentication involves collecting and storing the credentials, such as username and password of API consumers, and then requiring them to pass these credentials in the HTTP header for every API call made to the API server, these credentials are then checked for validity in order to authenticate the request.
+
+This method of authentication although simple to implement, is insecure because the credentials are not encrypted and can be exposed during transmission especially if used without HTTPS.
+
+Other forms of authentication such as API keys and Token-based authentication are more commonly adopted because of their improved security mechanisms in comparison to basic authentication, with token-based authentication being the most inherently secure.
+
+
+### API Keys
+
+API Keys are yet another simple and straightforward method of authentication. They are unique alphanumeric strings generated by the API provider and given to the developers who register their applications to include in their requests. The API Keys then act as a passcode, allowing the API to identify and authenticate registered clients when they make a request. 
+
+Who is using it: Many API providers use API keys, including big players like [Google Maps API](https://developers.google.com/maps/documentation/javascript/get-api-key) and [Stripe](https://docs.stripe.com/api/authentication). What's common among successful API providers using API keys is that they implement other measures to help their consumers keep their API keys secure. For instance, the Google Maps API platform makes provision for consumers to restrict the usage of their API keys to only certain applications, or IP addresses. This way a stolen API key cannot be used unless the request is coming from within the bounds of the restriction. 
+
+Stripe equally implements a similar feature that allows one to restrict the IP addresses that can make API calls using an API key. In addition, they implement key Rolling. Rolling a key revokes it and generates a replacement key.  Keys can be rolled manually or via a schedule that rolls the key after a certain time.
+
+![A diagram illustrating API key restriction by IP address](downloads/restricted-api-keys.png)
```

Comment:
@Shulammite-Aso the images should be in the `blog-assets` plus they don't seem to be a part of the pr

Reply:
Done this